packages/next/src/utils.ts

```diff
@@ -2567,10 +2512,13 @@ export function normalizeEdgeFunctionPath(
       appPathRoutesManifest[ogRoute] ||
       shortPath.replace(/(^|\/)(page|route)$/, '')
     ).replace(/^\//, '');
+  }
 
-    if (!shortPath || shortPath === '/') {
-      shortPath = 'index';
-    }
+  if (shortPath.startsWith('pages/')) {
+    shortPath = shortPath.replace(/^pages\//, '');
+  }
```

Comment:
This looks similar to https://github.com/vercel/vercel/pull/10630 but I don't see a test for this new behavior